analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -161,10 +170,13 @@ private void Verify(string fileName, ProjectType projectType, Action<IReadOnlyLi
                 TestHelper.CreateSonarProjectConfig(testRoot, projectType),
                 messages =>
                 {
-                    messages.Should().HaveCount(1);
-                    var info = messages.Single();
-                    info.FilePath.Should().Be(fileName);
-                    verifyReference(info.Reference);
+                    messages.Should().HaveCount(expectedMessageCount);
+                    var info = messages.FirstOrDefault();
+                    if (info != null)
+                    {
+                        info.FilePath.Should().Be(fileName);
+                        verifyReference(info.Reference);
+                    }
```

Comment:
There's implicit expectation that `expectedMessageCount` will be only 0 or 1. I'd prefer it this way to throw on `Single()` in case somebody breaks the expectation. He will have to rethink the test structure.
```suggestion
                    if (expectedMessageCount != 0)
                    {
                        var info = messages.Single();
                        info.FilePath.Should().Be(fileName);
                        verifyReference(info.Reference);
                    }
```